### PR TITLE
[PATCH v4] test: mem_perf: add memory performance test

### PR DESCRIPTION
**Pull request by psavol**
Added new test application that measures memory system
bandwidth.

### CODE REVIEW

File: test/performance/odp_mem_perf.c [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+	odp_barrier_wait(&global->barrier);
+
+	t1 = odp_time_local();
+
+	if (mode == 0) {
+		for (i = 0; i < num_round; i++)
+			memset(addr, thr + i, data_len);
+	} else {
+		for (i = 0; i < num_round; i++) {
+			if (i % 2 == 0)
```
Comment (jerinjacobk) [inline test/performance/odp_mem_perf.c:277]: Better to change to AND opertaion in fastpath instead of %. i.e i & 1

Comment (psavol) [reply test/performance/odp_mem_perf.c:277]: True. Fixed in v2.

**Comment by psavol** [comment]
v3: fixed shm names and added odp_shm_print_all() call

File: test/performance/odp_mem_perf.c [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+	odp_sys_info_print();
+
+	if (set_num_cpu(global))
+		return -1;
+
+	if (create_shm(global))
+		return -1;
+
+	/* Start workers */
+	start_workers(global, instance);
```
Comment (matiaselo) [inline test/performance/odp_mem_perf.c:418]: Return value note checked.

File: test/performance/odp_mem_perf.c [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+
+	/* Wait workers to exit */
+	odph_thread_join(global->thread_tbl, global->test_options.num_cpu);
+
+	print_stat(global);
+
+	free_shm(global);
```
Comment (matiaselo) [inline test/performance/odp_mem_perf.c:425]: Return value note checked.

**Review by matiaselo — APPROVED** [review]

**Pull request merged**